include/oneapi/dpl/numeric

```diff
@@ -68,4 +68,5 @@ using oneapi::dpl::transform_inclusive_scan;
 using oneapi::dpl::transform_reduce;
 } // namespace std
 
+#include "oneapi/dpl/internal/serial_numeric_impl.h"
```

Comment:
The code here seems to be very fragile, we have to put the include after the code injecting oneapi::dpl::reduce into std namespace.
The reason is if we put our impl of serial oneapi::dpl::reduce before the code injecting oneapi::dpl::reduce(parallel version), the target of "using oneapi::dpl::reduce" will be our impl and it will conflict with system std's reduce since they have exactly same prototype.
I am not sure whether do we have better approach, do you have good idea? @capatober  @rarutyun

Reply:
I put another comment for that. If we need to make serial algorithms available from the `oneapi::dpl::` namespace, we can also use feature presence macros (e.g. to add or not to add depending on having our own implementation or using standard algorithms)